destination/destination.go

```diff
@@ -0,0 +1,55 @@
+package destination
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	platform = "platform"
+	name     = "name"
+	os       = "OS"
+)
+
+type Simulator struct {
+	Platform string
+	Name     string
+	OS       string
+}
+
+func NewSimulator(destination string) (*Simulator, error) {
+	simulator := Simulator{}
+	destinationParts := strings.Split(destination, ",")
+
+	for _, part := range destinationParts {
+		keyAndValue := strings.Split(part, "=")
+		key := keyAndValue[0]
+		value := keyAndValue[1]
```

Comment:
If we follow [this cheatsheet](https://mokacoding.com/blog/xcodebuild-destination-options/) then I think yes:
* Platform should always be provided in the form of `platform=<platform>`.
* Name should always be provided in the form of `name=<name>`.
* OS should always be provided in the form of `OS=<os>` or be omitted.

So if any of these doesn't match, then the setup of the step is incorrect, and this part will fail.

Do you think it would make sense to add an extra check here to provide a better error message than a regular "array index out of bounds"? 😄

Reply:
IMO it makes sense, otherwise, the program will panic and the error message will not be helpful for our users